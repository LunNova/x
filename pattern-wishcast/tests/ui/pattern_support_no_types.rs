// SPDX-FileCopyrightText: 2025 LunNova
//
// SPDX-License-Identifier: MIT

//! Test error when enum declares pattern support but no pattern types are defined

use pattern_wishcast::pattern_wishcast;

pattern_wishcast! {
	enum Value is <P: PatternFields> = {
		HostValue { value: String },
		TupleValue { elements: Vec<Self> },
	};

	// Oops, forgot to define any pattern types!
}

fn main() {}
