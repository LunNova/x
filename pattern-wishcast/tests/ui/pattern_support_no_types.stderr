error: Enum `Value` declares pattern support with `is <P: ...>` but no pattern types are defined. Either: 1) Add pattern type declarations like `type FlexValue = Value is _;`, or 2) Remove the `is <P: ...>` declaration if you don't need pattern-based strictness.
  --> tests/ui/pattern_support_no_types.rs:9:1
   |
9  | / pattern_wishcast! {
10 | |     enum Value is <P: PatternFields> = {
11 | |         HostValue { value: String },
12 | |         TupleValue { elements: Vec<Self> },
...  |
16 | | }
   | |_^
   |
   = note: this error originates in the macro `pattern_wishcast` (in Nightly builds, run with -Z macro-backtrace for more info)
